flink-datastream/src/main/java/org/apache/flink/datastream/impl/context/DefaultKeyedTwoOutputNonPartitionedContext.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.datastream.impl.context;
+
+import org.apache.flink.datastream.api.common.Collector;
+import org.apache.flink.datastream.api.context.TwoOutputNonPartitionedContext;
+import org.apache.flink.datastream.api.function.TwoOutputApplyPartitionFunction;
+
+import java.util.Iterator;
+
+/**
+ * {@link TwoOutputNonPartitionedContext} for keyed operator. This will take care of the key context
+ * when apply to all keyed partitions.
+ */
+public class DefaultKeyedTwoOutputNonPartitionedContext<OUT1, OUT2>
+        extends DefaultTwoOutputNonPartitionedContext<OUT1, OUT2> {
```

Comment:
Not sure about further extending the class. Maybe we can simply use an argument to distinguish whether the stream is keyed or not.